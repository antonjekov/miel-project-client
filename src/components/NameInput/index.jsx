import React from 'react'
import styles from './index.module.css';
import { Form } from 'react-bootstrap';

function NameInput({ values, errors, handleChange }) {
    return (
        <Form.Group >
            <Form.Label>Name</Form.Label>
            <Form.Control className={styles['form-control']} type="text" placeholder="Enter name" name='name' value={values.name} onChange={handleChange} isInvalid={!!errors.name} isValid={values.name && !errors.name} />
            <Form.Control.Feedback type='invalid' >{errors.name}</Form.Control.Feedback>
            <Form.Control.Feedback type='valid'>Look's good</Form.Control.Feedback>
        </Form.Group>
    )
}

export default NameInput