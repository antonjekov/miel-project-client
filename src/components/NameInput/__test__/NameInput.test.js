import React from 'react'
import renderer from "react-test-renderer"
import {render, cleanup, screen, fireEvent} from "@testing-library/react"
import NameInput from '..'

describe('NameInput', ()=>{

    afterEach(cleanup)
    it('matches snapshot without error', () => {
        const values = {name: 'Anton'}
        const errors = {}
        const tree = renderer.create( <NameInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })
    
    it('matches snapshot with error', () => {
        const values = {name: 'Anton'}
        const errors = {name: 'Some error'}
        const tree = renderer.create( <NameInput values={values} errors={errors} handleChange={()=>{}}/>
        ).toJSON();
    
        expect(tree).toMatchSnapshot();
    })

})
