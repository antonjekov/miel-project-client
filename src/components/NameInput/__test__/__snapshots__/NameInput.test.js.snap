// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameInput matches snapshot with error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Name
  </label>
  <input
    className="form-control form-control is-invalid"
    name="name"
    onChange={[Function]}
    placeholder="Enter name"
    type="text"
    value="Anton"
  />
  <div
    className="invalid-feedback"
  >
    Some error
  </div>
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;

exports[`NameInput matches snapshot without error 1`] = `
<div
  className="form-group"
>
  <label
    className="form-label"
  >
    Name
  </label>
  <input
    className="form-control form-control is-valid"
    name="name"
    onChange={[Function]}
    placeholder="Enter name"
    type="text"
    value="Anton"
  />
  <div
    className="invalid-feedback"
  />
  <div
    className="valid-feedback"
  >
    Look's good
  </div>
</div>
`;
